packages/stainless/src/stl.ts

```diff
@@ -190,74 +256,146 @@ export class StlError extends Error {
   }
 }
 
+/** When thrown in an endpoint handler, responds with HTTP status code 400. */
 export class BadRequestError extends StlError {
+  /**
+   * @param response  optional data included in the body of the response JSON.
+   */
   constructor(response?: Record<string, any>) {
     super(400, { error: "bad request", ...response });
   }
```

Comment:
Actually, I feel like this would be a lot cleaner with something more like this:

```ts
export class BadRequestError extends StlError {
  statusCode = 400;
}
```
so we can just inherit the base constructor. I don't think the `error: "bad request"` type thing adds that much value; though maybe we could do this:
```ts
export class BadRequestError extends StlError {
  statusCode = 400;
  type = 'bad_request';
}
```
where then users can subclass like so:
```ts
export class InvalidParamsError extends BadRequestError {
  type = 'invalid_params';
}
```
and then get nicer custom errors in their SDKs etc (like we do for Increase, who have [thoughtful error types](https://increase.com/documentation/api#errors)).

By default / if unspecified, the statusCode should be `500`.

Reply:
I think so, ideally

Reply:
Haven't thought about it much